hw/xbox/nv2a/nv2a_pgraph.c

```diff
@@ -3083,6 +3134,23 @@ static void pgraph_bind_shaders(PGRAPHState *pg)
 
     glUseProgram(pg->shader_binding->gl_program);
 
+    /* Clipping regions */
+    for (i = 0; i < state.psh.window_clip_count; i++) {
+        uint32_t x = pg->regs[NV_PGRAPH_WINDOWCLIPX0 + i * 4];
+        GLuint xmin = GET_MASK(x, NV_PGRAPH_WINDOWCLIPX0_XMIN);
+        GLuint xmax = GET_MASK(x, NV_PGRAPH_WINDOWCLIPX0_XMAX);
+        uint32_t y = pg->regs[NV_PGRAPH_WINDOWCLIPY0 + i * 4];
+        GLuint ymin = pg->surface_shape.clip_height - 1 -
+                      GET_MASK(y, NV_PGRAPH_WINDOWCLIPX0_XMIN);
+        GLuint ymax = pg->surface_shape.clip_height - 1 -
+                      GET_MASK(y, NV_PGRAPH_WINDOWCLIPX0_XMAX);
+        if (pg->shader_binding->clip_region_loc[i] != -1) {
+            pgraph_apply_anti_aliasing_factor(pg, &xmin, &ymin);
+            pgraph_apply_anti_aliasing_factor(pg, &xmax, &ymax);
+            glUniform4i(pg->shader_binding->clip_region_loc[i], xmin, ymax, xmax, ymin);
```

Comment:
The order of these parameters is confusing, but it is because D3D and GL use different coordinate systems.
Are you going to add a comment?